[CI] FieldExtractorIT testByteFieldWithIntSubfield failing

**Build Scans:**
- [elasticsearch-periodic #8707 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/gvybfuc6etnxm)
- [elasticsearch-periodic #8707 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/7nvrmbtiet7ce)
- [elasticsearch-periodic #8707 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/zrr4r5co2sk5g)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.13.4#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT" -Dtests.method="testByteFieldWithIntSubfield" -Dtests.seed=A15AE52563A9DEDA -Dtests.bwc=true -Dtests.locale=vai-Vaii -Dtests.timezone=Hongkong -Druntime.java=24
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.FieldExtractorIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testByteFieldWithIntSubfield),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
columns: a list containing
        0: a map containing
       type: "integer"
       name: "byte_field"
        1: a map containing
       type: "integer"
       name: "byte_field.int"
 values: a list containing
        0: a list containing
          0: <99>
          1: expected <99> but was null
```

**Issue Reasons:**
- [8.19] 3 consecutive failures in test testByteFieldWithIntSubfield
- [8.19] 3 failures in test testByteFieldWithIntSubfield (1.9% fail rate in 158 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FieldAttribute.java
+++ b/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FieldAttribute.java
@@ -156,8 +156,9 @@ public class FieldAttribute extends TypedAttribute {
             // name starting with `$$`.
             if ((synthetic() || name().startsWith(SYNTHETIC_ATTRIBUTE_NAME_PREFIX)) == false) {
                 lazyFieldName = new FieldName(name());
+            } else {
+                lazyFieldName = new FieldName(Strings.hasText(parentName) ? parentName + "." + field.getName() : field.getName());
             }
-            lazyFieldName = new FieldName(Strings.hasText(parentName) ? parentName + "." + field.getName() : field.getName());
         }
         return lazyFieldName;
     }

